File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/dbo/error-log/error-log-lookup-model.ts
export interface IErrorLogLookupModel {
  errorLogID: number;
  errorTime: Date | string;
  userName: string;
  errorNumber: number;
  errorSeverity: number | null;
  errorState: number | null;
  errorProcedure: string;
  errorLine: number | null;
  errorMessage: string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class ErrorLogLookupModel implements IErrorLogLookupModel {
  errorLogID: number;
  errorTime: Date | string;
  userName: string;
  errorNumber: number;
  errorSeverity: number | null;
  errorState: number | null;
  errorProcedure: string;
  errorLine: number | null;
  errorMessage: string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<ErrorLogLookupModel>) {
    if (!!init) {
      Object.assign<ErrorLogLookupModel, Partial<ErrorLogLookupModel>>(this, init);
    }
  }
}

File: src/AdventureWorks.Application/DataEngine/Dbo/ErrorLog/Commands/Queries/GetErrorLogs/ErrorLogLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Dbo;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;

namespace AdventureWorks.Application.DataEngine.Dbo.ErrorLog.Queries.GetErrorLogs
{
    public partial class ErrorLogLookupModel : IHaveCustomMapping
    {
        public int ErrorLogID { get; set; }
        public DateTime ErrorTime { get; set; }
        public string UserName { get; set; }
        public int ErrorNumber { get; set; }
        public int? ErrorSeverity { get; set; }
        public int? ErrorState { get; set; }
        public string ErrorProcedure { get; set; }
        public int? ErrorLine { get; set; }
        public string ErrorMessage { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Dbo.ErrorLog, ErrorLogLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<ErrorLogLookupModel, BaseErrorLog>().IgnoreMissingDestinationMembers();
        }
    }
}
